src/clip.sol

```diff
@@ -196,10 +200,10 @@ contract Clipper is LibNote {
         // Could get this from rmul(Vat.ilks(ilk).spot, Spotter.mat()) instead,
         // but if mat has changed since the last poke, the resulting value will
         // be incorrect.
-        (PipLike pip, ) = spot.ilks(ilk);
+        (PipLike pip, uint256 mat) = spot.ilks(ilk);
         (bytes32 val, bool has) = pip.peek();
         require(has, "Clipper/invalid-price");
-        sales[id].top = rmul(rdiv(mul(uint256(val), BLN), spot.par()), buf);
+        sales[id].top = rmul(rdiv(max(rmul(lip, mat), mul(uint256(val), BLN)), spot.par()), buf);
```

Comment:
Btw only `mul(uint256(val), BLN)` should be divided by `par`, as `par` is a conversion factor from DAI to the references asset (i.e. `par` has units of `dollars / DAI`). `lip` is already `DAI / collateral` and `mat` is dimensionless.

Reply:
I'll fix this 👍